c2corg_ui/static/js/documentediting.js

```diff
@@ -33,7 +33,14 @@ app.documentEditingDirective = function() {
     scope: true,
     controller: 'appDocumentEditingController',
     controllerAs: 'editCtrl',
-    bindToController: true
+    bindToController: true,
+    link: function(scope, el, attrs, ctrl) {
+      // The directive executes after the dom is loaded but we have to wait until the expressions are evaluated
```

Comment:
max_steps ( = $(.step).length ) has to be evaluated and it's the number of .step classes that depend on ng-if="waypoint.waypoint_type", some have more or less steps and I wait until for this eval (not sure how to do better). If I don't do that, max_steps will first be initialized as 4 while it's real value will have changed to 5 in between. I've had this problem when changing waypoint type and max_steps was always equal to it's previous value, not the selected one!